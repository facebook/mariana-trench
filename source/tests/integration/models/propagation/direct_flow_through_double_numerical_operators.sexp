(method (private) "LFlow;.flow:()V"
 (
  LINE(1)
  (invoke-virtual (v0) "LSource;.source:()LData;")
  (move-result-object v1)

  LINE(2)
  (const-wide v0 3)
  (mul-double v2 v1 v0)

  LINE(3)
  (const-wide v0 1)
  (add-double v3 v2 v0)

  LINE(4)
  (const-wide v0 2)
  (div-double v4 v3 v0)

  LINE(5)
  (const-wide v0 1)
  (sub-double v5 v0 v4)

  LINE(6)
  (invoke-virtual (v0 v4) "LSink;.sink:(LData;)V")
  (return-void)
 )
)
