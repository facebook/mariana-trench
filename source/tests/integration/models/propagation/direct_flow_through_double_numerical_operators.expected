{
  "metadata" :
  {
    "@generated" : true
  },
  "models" :
  [
    {
      "issues" :
      [
        {
          "always_features" :
          [
            "via-numerical-operator"
          ],
          "position" :
          {
            "line" : 6,
            "path" : "Flow.java"
          },
          "rule" : 1,
          "sinks" :
          [
            {
              "call_position" :
              {
                "line" : 6,
                "path" : "Flow.java"
              },
              "callee" : "LSink;.sink:(LData;)V",
              "callee_port" : "Argument(1)",
              "distance" : 1,
              "kind" : "TestSink",
              "origins" :
              [
                "LSink;.sink:(LData;)V"
              ]
            }
          ],
          "sources" :
          [
            {
              "always_features" :
              [
                "via-numerical-operator"
              ],
              "call_position" :
              {
                "line" : 1,
                "path" : "Flow.java"
              },
              "callee" : "LSource;.source:()LData;",
              "callee_port" : "Return",
              "distance" : 1,
              "kind" : "TestSource",
              "local_features" :
              {
                "always_features" :
                [
                  "via-numerical-operator"
                ]
              },
              "local_positions" :
              [
                {
                  "line" : 2
                },
                {
                  "line" : 3
                },
                {
                  "line" : 4
                }
              ],
              "origins" :
              [
                "LSource;.source:()LData;"
              ]
            }
          ]
        }
      ],
      "method" : "LFlow;.flow:()V",
      "position" :
      {
        "line" : 0,
        "path" : "Flow.java"
      }
    }
  ]
}
